### PR TITLE
23620 - Add in year filter

### PR DESCRIPTION
**Pull request by seeker25**
*Issue #:*
https://github.com/bcgov/entity/issues/23620 

*Description of changes:*
Add in yearly filter for transactions, should smooth out the load on pay-api for the large transaction screen as well as the smaller client ones.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-3162-tcls9493.web.app

File: auth-web/src/composables/transactions-factory.ts [hunk 1]
```diff
@@ -115,18 +116,31 @@ export const useTransactions = () => {
     }
   }
 
-  const clearAllFilters = () => {
+  const clearAllFilters = (skipLoadTransactions = false) => {
     transactions.filters.filterPayload = { dateFilter: { startDate: '', endDate: '' } }
     transactions.filters.isActive = false
     transactions.filters.pageNumber = 1
-    loadTransactionList()
+    if (!skipLoadTransactions) {
```
Comment (seeker25) [inline auth-web/src/composables/transactions-factory.ts:123]: Fix double load issue

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-3162-tcls9493.web.app

File: auth-web/src/components/auth/account-settings/transaction/TransactionsDataTable.vue [hunk 2]
```diff
@@ -4,6 +4,8 @@
       v-show="showDatePicker"
       ref="datePicker"
       :reset="dateRangeReset"
+      :setStartDate="transactions.filters.filterPayload.dateFilter.startDate"
```
Comment (seeker25) [inline auth-web/src/components/auth/account-settings/transaction/TransactionsDataTable.vue:7]: only gets set once

File: auth-web/src/components/auth/account-settings/transaction/TransactionsDataTable.vue [hunk 3]
```diff
@@ -73,7 +75,7 @@
             filled
             hide-details
             :placeholder="'Date'"
-            :value="dateRangeSelected ? 'Custom' : ''"
+            :value="dateRangeSelected ? 'Custom' : '1 Year'"
```
Comment (seeker25) [inline auth-web/src/components/auth/account-settings/transaction/TransactionsDataTable.vue:78]: Custom disappears with clear (no text), also shows up after you click a date

**Review by ochiu — APPROVED** [review]

File: auth-web/src/composables/transactions-factory.ts [hunk 4]
```diff
@@ -115,18 +116,31 @@ export const useTransactions = () => {
...
+      loadTransactionList()
+    }
+  }
+
+  // We need this specific function to set the default search to one year, otherwise too many rows and loss of performance.
+  const defaultSearchToOneYear = () => {
+    transactions.filters.filterPayload = {
+      dateFilter: {
+        startDate: moment().subtract(1, 'year').format('YYYY-MM-DD'),
+        endDate: moment().add(1, 'day').format('YYYY-MM-DD')
```
Comment (seeker25) [inline auth-web/src/composables/transactions-factory.ts:133]: Just in case it sits for a bit and rolls over midnight somehow

**Pull request merged**